javascript-style-guide.md

```diff
@@ -34,24 +37,67 @@ avoid whitespace-oriented commits and diffs.
 
 Use JSCS to validate the format of your code. Your editor may offer JSCS
 integration, which you may find useful if you are not familiar with the style
-guide. You can also use our projects' Grunt interface.
+guide. You can also use our projects' Grunt or Gulp interfaces.
+
+# English
+
+## Spelling
 
-# Grammar
+Use real words for identifiers.
 
-## Use real words for identifiers.
+For instance, "element" is best written as "element", not "el", "elm", "elem",
+or "e".
 
-"element" is best written as "element", not "el", "elm", "elem", or "e".
+For your convenience, the following words also have a preferred spelling:
+
+Word      | Best written as | Known to be written as
+----------|-----------------|-----------------------
+directory | directory       | dir
+error     | error           | err, e
+index     | index           | idx, i
+request   | request         | req
+response  | response        | res
```

Comment:
Agreed, I'd like not to have to focus on this point, it just seems to be a rough point for newcomers, as abbreviations are so infectious in the programming field.

We could simplify the style guide (and focus on the points that matter, like clarity) by offloading matters like formatting and spelling to JSCS and [jscs-spellcheck](https://github.com/jscs-dev/jscs-spellcheck). Most of Crockford's style guide is just formatting anyway, so offloading also kills that dependency.